tests/src/Api/SearchApiTest.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\Tests\EuropaSearchClient\Api;
+
+use GuzzleHttp\Psr7\Response;
+use OpenEuropa\EuropaSearchClient\Model\Document;
+use OpenEuropa\EuropaSearchClient\Model\QueryLanguage;
+use OpenEuropa\EuropaSearchClient\Model\Search;
+use OpenEuropa\Tests\EuropaSearchClient\Traits\ClientTestTrait;
+use OpenEuropa\Tests\EuropaSearchClient\Traits\InspectTestRequestTrait;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * @coversDefaultClass \OpenEuropa\EuropaSearchClient\Api\SearchApi
+ */
+class SearchApiTest extends TestCase
+{
+    use ClientTestTrait;
+    use InspectTestRequestTrait;
+
+    /**
+     * @covers ::search
+     * @dataProvider providerTestSearch
+     *
+     * @param array $clientConfig
+     * @param array $responses
+     * @param mixed $expectedResult
+     */
+    public function testSearch(array $clientConfig, array $responses, $expectedResult): void
+    {
+        $actualResult = $this->getTestingClient($clientConfig, $responses, [$this, 'inspectRequest'])
+            ->search(
+                'Programme managers',
+                ['en', 'de'],
+                ['term' => ['DMAKE_ES_EVENT_TYPE_NAME' => 'ADOPTION_DISTRIBUTE']],
+                'es_SortDate',
+                'DESC',
+                2,
+                5,
+                '{w+}',
+                150,
+                '21edswq223rews'
+            );
+        $this->assertEquals($expectedResult, $actualResult);
+    }
+
+    /**
+     * @param RequestInterface $request
+     */
+    public function inspectRequest(RequestInterface $request): void
+    {
+        $this->assertEquals('http://example.com/search?apiKey=foo&database=bar&text=Programme+managers&sessionToken=21edswq223rews&pageNumber=2&pageSize=5&highlightRegex=%7Bw%2B%7D&highlightLimit=150', $request->getUri());
+        $this->inspectBoundary($request);
+        $parts = $this->getMultiParts($request);
+        $this->assertCount(3, $parts);
+        $this->inspectPart($parts[0], 'application/json', 'languages', 11, '["en","de"]');
+        $this->inspectPart($parts[1], 'application/json', 'query', 59, '{"term":{"DMAKE_ES_EVENT_TYPE_NAME":"ADOPTION_DISTRIBUTE"}}');
+        $this->inspectPart($parts[2], 'application/json', 'sort', 38, '{"field":"es_SortDate","order":"DESC"}');
+    }
+
+    /**
+     * @see self::testSearch()
+     */
+    public function providerTestSearch(): array
+    {
+        return [
+            'simple search' => [
+                [
+                    'apiKey' => 'foo',
+                    'database' => 'bar',
+                    'searchApiEndpoint' => 'http://example.com/search',
+                ],
+                [
+                    new Response(200, [], json_encode([
```

Comment:
This response is clearly better stored in a fixture json file.

Reply:
done